### ISSUE TITLE
Default to ANSI dialect

### ISSUE THREAD

**Issue by boldidev** [sqlfluff/sqlfluff#7278]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

When the user does not specify a dialect, assume the ANSI dialect.

#### Expected Behavior

1. User installs sqlfluff
2. User lints project with sqlfluff 
   ```sh
   sqlfluff lint
   ```
3. `All Finished 📜 🎉!`

#### Actual Behavior

1. User installs sqlfluff
2. User lints project with sqlfluff 
   ```sh
   sqlfluff lint
   ```
3. ```log
   User Error: No dialect was specified. You must configure a dialect or specify one on the command line using --dialect after the command. Available dialects:
   ansi, athena, bigquery, clickhouse, databricks, db2, doris, duckdb, exasol, flink, greenplum, hive, impala, mariadb, materialize, mysql, oracle, postgres, redshift, snowflake, soql, sparksql, sqlite, starrocks, teradata, trino, tsql, vertica
   ```


### Use case

Allows users to start linting with zero configuration.
Since it is just a default, when a config (e.g. `.sqlfluff` file) is present, it will use the correct dialect specified in there.

It is currently impossible to have SQLFluff work with zero-configuration, while also respecting the config file.

[This issue](https://github.com/mfussenegger/nvim-lint/issues/819) on the nvim-lint project (it provides in-editor lint results for neovim) is a result of no default behavior. The project tried to make it work without extra config needed, but the CLI `--dialect` option always overrides what is present in the config file.

### Dialect

-

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by WittierDinosaur** [comment]
As per the discussion in #7282, closing, as this is a behaviour change that we don't want at this time

**Comment by boldidev** [comment]
(quoting from #7282)

> It makes the dialect config more obscure for the user when they're setting up their project

I do wholly agree with this point. Had it defaulted to ANSI I would have not looked into the `sqlfluff` configuration in the foreseeable future. I can 100% see the lack of default as a feature.

So, as an alternative: Is the idea of introducing a default dialect flag worth considering? Say, something like `--default-dialect=ansi`? Possibly an environment variable instead? This would allow other tools to work out of the box, without obscuring the configuration file when using the tool directly.

(This still technically hides the configuration file a little when using sqlfluff through other tools that opt to add the default dialect flag.)
